src/main/java/com/mattermost/integration/figma/api/figma/notification/service/FileNotificationServiceImpl.java

```diff
@@ -87,16 +90,35 @@ public void subscribeToFileNotification(InputPayload inputPayload) {
         String botAccessToken = inputPayload.getContext().getBotAccessToken();
         String teamId = inputPayload.getValues().getTeamId();
 
-        deleteExistingFileCommentWebhook(teamId, mmSiteUrl, botAccessToken);
-
+        String currentTeamWebhookId = kvService.get(TEAM_WEBHOOK_PREFIX.concat(teamId), mmSiteUrl, botAccessToken);
         String accessToken = getToken(inputPayload);
 
-        HttpEntity<FileCommentNotificationRequest> request = createFileCommentNotificationRequest(inputPayload, accessToken);
+        Optional<Webhook> webhookOptional = tryToCreateWebhook(inputPayload, accessToken, teamId);
+        if (webhookOptional.isPresent()) {
+            deleteSingleFileCommentWebhook(currentTeamWebhookId, teamId, mmSiteUrl, botAccessToken);
+            saveNecessaryDataToKv(inputPayload, webhookOptional.get());
+            return;
+        }
+
+        if (!StringUtils.isBlank(currentTeamWebhookId)) {
+            return;
+        }
+
+        throw new FigmaCannotCreateWebhookException();
+    }
+
+    private Optional tryToCreateWebhook(InputPayload payload, String accessToken, String teamId) {
+        HttpEntity<FileCommentNotificationRequest> request = createFileCommentNotificationRequest(payload, accessToken);
         log.debug("File notification request : " + request);
         log.info("Sending comment request for team with id: " + teamId);
-        ResponseEntity<String> stringResponseEntity = figmaRestTemplate.postForEntity(BASE_WEBHOOK_URL, request, String.class);
-        Webhook webhook = (Webhook) jsonUtils.convertStringToObject(stringResponseEntity.getBody(), Webhook.class).get();
-        saveNecessaryDataToKv(inputPayload, webhook);
+        ResponseEntity<String> stringResponseEntity;
+        try {
+            stringResponseEntity = figmaRestTemplate.postForEntity(BASE_WEBHOOK_URL, request, String.class);
+        } catch (RuntimeException e) {
+            return Optional.empty();
+        }
+
+        return  Optional.of(jsonUtils.convertStringToObject(stringResponseEntity.getBody(), Webhook.class).orElse(null));
```

Comment:
.orElse(null) I guess it's a little bit tricky to return null in method that returns Optional . What about Optional.empty?